mne_bids/read.py

```diff
@@ -830,15 +830,42 @@ def get_head_mri_trans(bids_path, extra_params=None, t1_bids_path=None,
     meg_bids_path.update(datatype='meg', suffix='meg')
 
     # Get the sidecar file for MRI landmarks
-    match_bids_path = meg_bids_path if t1_bids_path is None else t1_bids_path
-    t1w_path = _find_matching_sidecar(
-        match_bids_path, suffix='T1w', extension='.nii.gz')
+    t1w_bids_path = (
+        (meg_bids_path if t1_bids_path is None else t1_bids_path)
+        .copy()
+        .update(
+            datatype='anat',
+            suffix='T1w',
+            task=None
+        )
+    )
     t1w_json_path = _find_matching_sidecar(
-        match_bids_path, suffix='T1w', extension='.json')
+        bids_path=t1w_bids_path, extension='.json', on_error='ignore'
+    )
+    del t1_bids_path
```

Comment:
why are you deleting this variable here?

Reply:
to make sure we don't use it below I guess

Reply:
We have `t1w_bids_path` and `t1_bids_path` when we reach this line; I want to avoid accidental confusion, so I'm deleting the one that's not needed anymore :) Old habit!